app/models/organization_resolver.rb

```diff
@@ -48,6 +49,25 @@ def organization_tag_names
     @organization&.tags || []
   end
 
+  def entities
+    [
+      @organization,
+      @email,
+      @location,
+      @organization_name,
+      @phone_number,
+      @website
+    ]
+  end
+
+  def source_names
+    entities.reduce([]) do |memo, entity|
+      next entity if entity.nil?
+      name = entity.versions.last.actor.import.source.name
+      memo.include?(name) ? memo : memo << name
+    end
```

Comment:
I might be missing something here, but looks like this function can be simplified as something like:
```ruby
entities.compact.map(&:versions.last.actor.import.source.name).uniq
```